build.gradle

```diff
@@ -201,6 +201,11 @@ task engineJar(type: Jar) {
                 "Implementation-Vendor": "Mozilla Foundation",
                 "Implementation-URL": "http://www.mozilla.org/rhino",
                 "Automatic-Module-Name": "org.mozilla.rhino.engine"
+                "Bundle-ManifestVersion": "2",
+                "Bundle-SymbolicName": "org.mozilla.rhino.engine",
+                "Bundle-Version": project.version.replaceAll("-.*", ""),
+                "Export-Package": "org.mozilla.javascript.engine",
+                "Import-Package": "javax.script,org.mozilla.javascript"
```

Comment:
I would not declare an `Import-Package`, and instead leave the default (automatic) behaviour.

Reply:
this is an OSGi import. If you do not declare it the bundle will not work/ throw an ClassNotFoundExcaption because bundle does not declare that is needs it.

Reply:
That's weird. I do not use OSGi myself, however from https://bnd.bndtools.org/heads/import_package.html:

> The `Import-Package` header lists the packages that are required by the contained packages. The default for this header is `*`, resulting in importing all referred packages. This header therefore rarely has to be specified.

Reply:
yep. bnd is a tool that calculates  die imports. if you say to bnd do `*` then it will habe as result in the manifest:

`Import-Package javax.script,org.mozilla.javascript`

@juergen-albert (OSGi Working Group Steering Committee)

could you please review this from the OSGi piunt of view?

Reply:
What @stbischof wrote is correct. bnd is a build tool, that can calculate the Imports for you with proper Version ranges. The same goes for the Export Packages. OSGi requires all the packages to be named separately and can not interpret the *.

bnd even has a dedicated gradle plugin that can automatically calculate a Manifest for your, not only putting in proper OSGi Headers, but can also automatically handle things in respect to Java Module System.

I'm not that firm with gradle, but you can find an example of the gradle plugin usage here: https://github.com/osgi/osgi-test/tree/main/examples/osgi-test-example-gradle